command/loadbot/execution.go

```diff
@@ -102,29 +111,44 @@ type Loadbot struct {
 }
 
 func NewLoadbot(cfg *Configuration) *Loadbot {
-	return &Loadbot{
+	loadbot := &Loadbot{
 		cfg: cfg,
 		metrics: &Metrics{
 			TotalTransactionsSentCount: 0,
 			FailedTransactionsCount:    0,
 			TransactionDuration: ExecDuration{
 				blockTransactions: make(map[uint64]uint64),
 			},
-			ContractMetrics: ContractMetricsData{
-				ContractDeploymentDuration: ExecDuration{
-					blockTransactions: make(map[uint64]uint64),
-				},
-				ContractGasMetrics: &BlockGasMetrics{
-					Blocks:        make(map[uint64]GasMetrics),
-					BlockGasMutex: &sync.Mutex{},
-				},
-			},
 			GasMetrics: &BlockGasMetrics{
-				Blocks:        make(map[uint64]GasMetrics),
-				BlockGasMutex: &sync.Mutex{},
+				Blocks: make(map[uint64]GasMetrics),
 			},
 		},
 	}
+
+	if loadbot.needsContractMetrics() {
+		loadbot.initContractMetrics()
+	}
+
+	return loadbot
+}
+
+// initContractMetrics initializes contract metrics for
+// the loadbot instance
+func (l *Loadbot) initContractMetrics() {
+	l.metrics.ContractMetrics = &ContractMetricsData{
+		ContractDeploymentDuration: ExecDuration{
+			blockTransactions: make(map[uint64]uint64),
+		},
+		ContractGasMetrics: &BlockGasMetrics{
+			Blocks: make(map[uint64]GasMetrics),
+		},
+	}
+}
+
+func (l *Loadbot) needsContractMetrics() bool {
+	return l.cfg.GeneratorMode == deploy ||
+		l.cfg.GeneratorMode == erc20 ||
+		l.cfg.GeneratorMode == erc721
 }
```

Comment:
Why separate to 2 methods `initContractMetrics` and `needsContractMetrics`?
How about adding a keyword to `initContractMetrics` so that it may initialize contract metrics
`mightInitContractMetrics`, `initContractMetricsIfNeeded`

Reply:
Fixed 👍 

[773dcd1](https://github.com/0xPolygon/polygon-edge/pull/497/commits/773dcd1fd3f245769c16491b3f09df9188aec227)